debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/buffered/BufferedLogMinerStreamingChangeEventSource.java

```diff
@@ -674,6 +685,16 @@ private Scn calculateNewStartScn(Scn startScn, Scn endScn, Scn maxCommittedScn)
         }
     }
 
+    /**
+     * Marks the specific transaction as skipped.
+     *
+     * @param transactionId the transaction identifier, should not be {@code null}
+     * @param eventScn the transaction's starting system change number, should not be {@code null}
+     */
+    protected void markTransactionSkipped(String transactionId, Scn eventScn) {
+        getProcessedTransactionsCache().put(transactionId, eventScn.toString());
```

Comment:
@Naros I wonder if we just want to put the transaction into processed cache. Would not it be better for potential future debugging also record the reason for putting into the cache -either finalized or skipped?

Reply:
@Naros Any opinion?

Reply:
That's a good question, :thinking: 

My only reservation would be that it requires the introduction of another marshaller to serialize an object as the value in the Map rather than a native String, which might add some overhead, tho it might be negligible. So, we either do that or use a separate Map for this _skipped_ use case and let the code differentiate the two, thereby avoiding the marshaller requirement.

Do you have any opinion? :wink:

Reply:
Ok, replaced cache value with an object. Talk about boilerplate ;-)